specs/decimal/decimal.md

```diff
@@ -197,66 +339,307 @@ Windows App SDK provides a native language decimal data type for C++ as the
+{
+/// Return value as a Win32 DECIMAL structure.
+inline DECIMAL to_DECIMAL(winrt::Microsoft::Windows::Foundation::DecimalValue const& value);
+
+/// Return value as a WinRT DecimalValue structure.
+inline winrt::Microsoft::Windows::Foundation::DecimalValue to_DecimalValue(DECIMAL const& value);
+}
+
+//----------------------------------------------------------
+// WinRT DecimalValue operators
+
+// C++/WinRT generates bool operator==(DecimalValue const& left, DecimalValue const& right)
+
+// C++/WinRT generates bool operator!=(DecimalValue const& left, DecimalValue const& right)
+
+inline bool operator<(
+    winrt::Microsoft::Windows::Foundation::DecimalValue const& left,
+    winrt::Microsoft::Windows::Foundation::DecimalValue const& right);
+
+inline bool operator<=(
+    winrt::Microsoft::Windows::Foundation::DecimalValue const& left,
+    winrt::Microsoft::Windows::Foundation::DecimalValue const& right);
+
+inline bool operator>(
+    winrt::Microsoft::Windows::Foundation::DecimalValue const& left,
+    winrt::Microsoft::Windows::Foundation::DecimalValue const& right);
+
+inline bool operator>=(
+    winrt::Microsoft::Windows::Foundation::DecimalValue const& left,
+    winrt::Microsoft::Windows::Foundation::DecimalValue const& right);
+
+#endif // defined(WINRT_Microsoft_Windows_Foundation_H) && !defined(__WINDOWSAPPSDK_WINRT_M_W_F_DECIMAL_)
+
+//----------------------------------------------------------------------
+// C++ decimal support
+
+#if defined(WINRT_Microsoft_Windows_Foundation_H) && defined(__WindowsAppSDK_Microsoft_Windows_Foundation_decimal_) && !defined(__WINDOWSAPPSDK_CPP_M_W_F_DECIMAL_)
+#define __WINDOWSAPPSDK_CPP_M_W_F_DECIMAL_
+
+//----------------------------------------------------------
+// WinRT DecimalValue <-> C++ decimal conversions
+
+namespace winrt::Microsoft::Windows::Foundation
+{
+/// Return true if value is valid.
+constexpr bool is_valid(winrt::Microsoft::Windows::Foundation::DecimalValue const& value);
+
+/// Return value as a C++ decimal object.
+inline ::Microsoft::Windows::Foundation::decimal to_decimal(winrt::Microsoft::Windows::Foundation::DecimalValue const& value);
+
+/// Return value as a WinRT DecimalValue structure.
+inline winrt::Microsoft::Windows::Foundation::DecimalValue to_DecimalValue(::Microsoft::Windows::Foundation::decimal const& value);
+}
+
+#endif // defined(WINRT_Microsoft_Windows_Foundation_H) && defined(__WindowsAppSDK_Microsoft_Windows_Foundation_decimal_) && !defined(__WINDOWSAPPSDK_CPP_M_W_F_DECIMAL_)
+```
+
+# 9. C# API
+
+C# provides rich support via its [Decimal struct](https://learn.microsoft.com/dotnet/api/system.decimal).
+Windows App SDK adds a small C# API in an assembly for interop with `DecimalValue`.
+
+```c#
+namespace Microsoft.Windows.Foundation
+{
+    public static class DecimalExtensions
+    {
+        /// Return a WinRT DecimalValue structure.
+        public static Microsoft.Windows.Foundation.DecimalValue ToDecimalValue(this decimal d);
```

Comment:
How about `ToWindowsDecimal` ? "Decimal" as a standalone noun still seems to mean "the .NET Decimal type"

Reply:
Maybe make this the other way around?

```c#
public static decimal ToDecimal(this Microsoft.Windows.Foundation.DecimalValue value);
```

So you say:

```c#
DecimalValue dv = /* ... */;
decimal d = dv.ToDecimal();
```

Instead of the strange:

```c#
DecimalValue dv = /* ... */;
decimal d;
d = d.FromDecimalValue(d, dv);
```

Reply:
RECOMMEND: Change to `ToDecimal`

i.e. `returntype To<returntype>(sourcetype)`